src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -147,16 +165,23 @@ class IOUAmountPage extends React.Component {
         // Backspace button is pressed
         if (key === '<' || key === 'Backspace') {
             if (this.state.amount.length > 0) {
-                this.setState(prevState => ({
-                    amount: prevState.amount.slice(0, -1),
-                }));
+                this.setState((prevState) => {
+                    const selectionStart = prevState.selection.start === prevState.selection.end ? prevState.selection.start - 1 : prevState.selection.start;
+                    const amount = `${prevState.amount.substring(0, selectionStart)}${prevState.amount.substring(prevState.selection.end)}`;
+                    const selection = this.getNewSelection(prevState.selection, prevState.amount.length, amount.length);
+                    return {amount, selection};
+                });
             }
             return;
         }
 
         this.setState((prevState) => {
-            const amount = this.addLeadingZero(`${prevState.amount}${key}`);
-            return this.validateAmount(amount) ? {amount: this.stripCommaFromAmount(amount)} : prevState;
+            const amount = this.addLeadingZero(`${prevState.amount.substring(0, prevState.selection.start)}${key}${prevState.amount.substring(prevState.selection.end)}`);
+            if (!this.validateAmount(amount)) {
+                return prevState;
+            }
+            const selection = this.getNewSelection(prevState.selection, prevState.amount.length, amount.length);
```

Comment:
> @eVoloshchak bug on mWeb Safari - cursor moves to end on inserting a number in middle.

Oh wow, that's weird. It is caused by this [commit](https://github.com/Expensify/App/pull/10147/commits/f2e16898d653d7b1cee7b97509571f65d48575a5), for some reason, Safari doesn't like textinput's value stored in state. 
[Comment](https://github.com/Expensify/App/blob/90b5292c89ff4516d3f4902d728dcabb67ce3786/src/components/TextInput/BaseTextInput.js#L34) in BaseTextInput.js suggests we should keep the value in state
> Value should be kept in state for the autoGrow feature to work

And for Safari to work, we use value from props, instead of from state (and apply the same logic for `defaultValue` as [here](https://github.com/Expensify/App/blob/90b5292c89ff4516d3f4902d728dcabb67ce3786/src/components/TextInput/BaseTextInput.js#L23))
```diff
- value={this.state.value}
+ value={this.props.value || this.props.defaultValue || ''}
```

I've updated the PR

https://user-images.githubusercontent.com/9059945/183156453-c3ea5052-9523-483b-b395-ac7462e57c46.mp4

Reply:
> Is there anyway to make it work on safari whilst using state as the value?

There is. It seemed logical to store `selection` in state the same way we store `value` and turns out it resolves the issue for Safari.

https://user-images.githubusercontent.com/9059945/183266862-0fe464c5-4e72-4cbd-ada3-73fd7e3ed2b3.mp4